backend/LexBoxApi/Jobs/RetryEmailJob.cs

```diff
@@ -0,0 +1,51 @@
+﻿using LexBoxApi.Services;
+using MimeKit;
+using Quartz;
+
+namespace LexBoxApi.Jobs;
+
+public class RetryEmailJob(EmailService emailService) : LexJob
+{
+    public static async Task Queue(ISchedulerFactory schedulerFactory,
+        MimeMessage email,
+        int retryCount = 3,
+        int retryWaitSeconds = 5 * 60,
+        CancellationToken cancellationToken = default)
+    {
+        var memory = new MemoryStream();
+        await email.WriteToAsync(memory, cancellationToken);
+        var serializedEmail = Convert.ToBase64String(memory.ToArray());
+        await QueueJob(schedulerFactory,
+            Key,
+            new JobDataMap {
+                { nameof(SerializedEmail), serializedEmail },
+                { nameof(RetryCount), retryCount },
+                { nameof(RetryWaitSeconds), retryWaitSeconds },
+            },
+            cancellationToken);
+    }
+
+    public static JobKey Key { get; } = new("RetryEmailJob", "RetryingJobs");
+    public string? SerializedEmail { get; set; }
+    public int RetryCount { get; set; }
+    public int RetryWaitSeconds { get; set; }
+
+    protected override async Task ExecuteJob(IJobExecutionContext context)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(SerializedEmail, "email");
```

Comment:
Shouldn't we throw when we encode it?

Reply:
This is to guard against something messing up the database while the job was queued. We can't realistically check for every condition, but we can at least check for the case where the row got zeroed out or similar.